<?xml version="1.0"?>
<launch>

    <!-- Launch PID Interpreter -->
    <node pkg="wamv_control" type="pid_interpreter" name="pid_interpreter_node" />

    <!-- Launch heading PID controller -->
    <node pkg="pid" type="controller" name="heading_controller" >
        <param name="Kp" value="1.0" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.0" />
        <param name="upper_limit" value="127" />
        <param name="lower_limit" value="-127" />
        <param name="topic_from_controller" value="heading_control_effort"/>
        <!-- NEED TO SET topic_from_plant VALUE TO WAMV CURRENT GPS HEADING topic -->
        <!--param name="topic_from_plant" value="heading?"/-->
        <param name="setpoint_topic" value="heading_to_target"/>
        <param name="pid_enable_topic" value="pid_enable"/>

        <!-- <param name="windup_limit" value="10" /> -->
        <!-- <param name="max_loop_frequency" value="100.0" /> -->
        <!-- <param name="min_loop_frequency" value="100.0" /> -->
   </node>

    <!-- Launch speed PID controller -->
    <node pkg="pid" type="controller" name="speed_controller" >
        <param name="Kp" value="1.0" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.0" />
        <param name="upper_limit" value="127" />
        <param name="lower_limit" value="-127" />
        <param name="topic_from_controller" value="speed_control_effort"/>
        <!-- NEED TO SET topic_from_plant VALUE TO WAMV CURRENT SPEED topic AND SET setpoint_topic TO SETPOINT OUTPUT FROM VELOCITY PLANNING -->
        <!--param name="topic_from_plant" value="speed?"/-->
        <!--param name="setpoint_topic" value="set_velocity"/-->
        <param name="pid_enable_topic" value="pid_enable"/>

        <!-- <param name="windup_limit" value="10" /> -->
        <!-- <param name="max_loop_frequency" value="100.0" /> -->
        <!-- <param name="min_loop_frequency" value="100.0" /> -->
   </node>
        <!-- <remap from="/navsat" to="/an_device/NavSatFix"/> -->
        <!-- <remap from="/imu" to="/an_device/Imu"/> -->
        <!-- <remap from="/gps" to="/gps/odom"/> -->
    <!-- </node> -->

</launch>
